src/GitLink/Linker.cs

```diff
@@ -23,234 +26,81 @@ namespace GitLink
 
-            exitCode = exitCode ?? -1;
-
-            if (context.ErrorsAsWarnings && exitCode != 0)
-            {
-                Log.Info("One or more errors occurred, but treating it as warning instead");
-
-                exitCode = 0;
-            }
-
-            return exitCode.Value;
-        }
-
-        private static bool LinkProject(Context context, Project project, string pdbStrFile, string shaHash, string pathPdbDirectory = null)
-        {
-            Argument.IsNotNull(() => context);
-            Argument.IsNotNull(() => project);
-
-            try
-            {
-                var projectName = project.GetProjectName();
-
-                Log.Info("Handling project '{0}'", projectName);
-
-                Log.Indent();
-
-                var compilables = project.GetCompilableItems().Select(x => x.GetFullFileName()).ToList();
-
-                var outputPdbFile = project.GetOutputPdbFile();
-                var projectPdbFile = pathPdbDirectory != null ? Path.Combine(pathPdbDirectory, Path.GetFileName(outputPdbFile)) : Path.GetFullPath(outputPdbFile);
-                var projectSrcSrvFile = projectPdbFile + ".srcsrv";
-
-                var srcSrvContext = new SrcSrvContext
-                {
-                    Revision = shaHash,
-                    RawUrl = context.Provider.RawGitUrl,
-                    DownloadWithPowershell = context.DownloadWithPowershell
-                };
-
-                if (!File.Exists(projectPdbFile))
-                {
-                    Log.Warning("No pdb file found for '{0}', is project built in '{1}' mode with pdb files enabled? Expected file is '{2}'", projectName, context.ConfigurationName, projectPdbFile);
-                    return false;
-                }
-
-                if (!context.SkipVerify)
+                if (!options.SkipVerify)
                 {
                     Log.Info("Verifying pdb file");
 
-                    var missingFiles = project.VerifyPdbFiles(compilables, projectPdbFile);
+                    var missingFiles = pdb.FindMissingOrChangedSourceFiles();
                     foreach (var missingFile in missingFiles)
                     {
-                        Log.Warning("Missing file '{0}' or checksum '{1}' did not match", missingFile.Key, missingFile.Value);
+                        Log.Warning($"File \"{missingFile}\" missing or changed since the PDB was compiled.");
                     }
                 }
 
-                if(!srcSrvContext.RawUrl.Contains("%var2%") && !srcSrvContext.RawUrl.Contains("{0}"))
+                string commitId;
+                commitId = repository.Head.Commits.First().Sha;
+
+                string rawUrl = provider.RawGitUrl;
+                if (!rawUrl.Contains("%var2%") && !rawUrl.Contains("{0}"))
                 {
-                    srcSrvContext.RawUrl = string.Format("{0}/{{0}}/%var2%", srcSrvContext.RawUrl);
+                    rawUrl = string.Format("{0}/{{0}}/%var2%", rawUrl);
```

Comment:
Agreed. Although at tip this particular line doesn't exist any more. So nothing to change here.